src/common/bpf.rs

```diff
@@ -10,6 +10,71 @@ pub struct BPF {
 #[cfg(not(feature = "bpf"))]
 pub struct BPF {}
 
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Hash)]
+#[cfg(feature = "bpf")]
+pub enum ProbeType {
+    Kernel,
+    User,
+    Tracepoint,
+}
+
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Hash)]
+#[cfg(feature = "bpf")]
+pub enum ProbeLocation {
+    Entry,
+    Return,
+}
+
+// Define a probe.
+#[derive(Clone, Debug, Eq, PartialEq, Hash)]
+#[cfg(feature = "bpf")]
+pub struct FunctionProbe {
```

Comment:
Why not just call this a `Probe`. We might also probe tracepoints, so I'm not sure that calling it a `FunctionProbe` makes sense.

Reply:
Renamed to Probe, yeah Tracepoints are actually supported here.